tests/functional/override_database/test_override_database.py

```diff
@@ -0,0 +1,252 @@
+            identifier=self.check_caps(project, "seed")
+        ),
+          project.adapter.Relation.create(
+            schema=project.test_schema,
+            identifier=self.check_caps(project, "view_1")
+        ),
+          project.adapter.Relation.create(
+            database=ALT_DATABASE,
+            schema=project.test_schema,
+            identifier=self.check_caps(project, "view_2")
+        ),
+          project.adapter.Relation.create(
+            schema=project.test_schema,
+            identifier=self.check_caps(project, "view_3")
+        ),
+          project.adapter.Relation.create(
+            database=ALT_DATABASE,
+            schema=project.test_schema,
+            identifier=self.check_caps(project, "view_4")
+        )
+      ])
+
+  def test_snwoflake_database_override(self, project, clean_up):
+    self.run_database_override(project)
+
+
+class BaseProjectModelOverrideSnowflake(BaseOverrideDatabaseSnowflake):
+  def run_database_override(self, project):
+        run_dbt(["seed"])
+        result = run_dbt(["run"])
+        assert len(result) == 4
+        # breakpoint()
+        check_relations_equal_with_relations(project.adapter, [
+              project.adapter.Relation.create(
+                schema=project.test_schema,
+                identifier=self.check_caps(project, "seed")
+            ),
+              project.adapter.Relation.create(
+                database=ALT_DATABASE,
+                schema=project.test_schema,
+                identifier=self.check_caps(project, "view_1")
+            ),
+              project.adapter.Relation.create(
+                database=ALT_DATABASE,
+                schema=project.test_schema,
+                identifier=self.check_caps(project, "view_2")
+            ),
+              project.adapter.Relation.create(
+                schema=project.test_schema,
+                identifier=self.check_caps(project, "view_3")
+            ),
+              project.adapter.Relation.create(
+                database=ALT_DATABASE,
+                schema=project.test_schema,
+                identifier=self.check_caps(project, "view_4")
+            )
+          ])
+
+
+
+class TestProjectModelOverrideSnowflake(BaseProjectModelOverrideSnowflake):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "vars": {
+                "alternate_db": ALT_DATABASE,
+            },
+            "models": {
+              "database": ALT_DATABASE,
+                "test": {
+                    "subfolder": {
+                        "database": "{{ target.database }}"
+                    }
+                }
+            },
+            "vars": {
+                "alternate_db": ALT_DATABASE,
+            },
+        }
+
+    def test_snowflake_database_override(self, project, clean_up):
+      self.run_database_override(project)
+
+# After second thought pretty sure this is only for bigquery correct? dataset = project
```

Comment:
after looking back at original test which has the setup for this one but no actual implementation I believe this may be just for bigquery where dataset and project are interchangable, didn't see anything relevant in docs either in regards to snowflake.

Reply:
We should clean it up if it isn't running an actual test.